cli/src/test/java/org/jboss/pnc/bacon/cli/AppTest.java

```diff
@@ -76,90 +85,88 @@ void testJSONDisabled() throws Exception {
         File configYaml = new File(pncClasses.getParentFile().getParentFile().getParentFile(), PNC_TEST_CLASSES);
 
         String text = tapSystemErr(
-                () -> assertEquals(
-                        1,
-                        new App().run(new String[] { "-p", configYaml.toString(), "-v", "pnc", "build", "get", "0" })));
-        assertTrue(text.contains("JSON command is enabled: false"));
+                () -> assertThat(
+                        new App().run(new String[] { "-p", configYaml.toString(), "-v", "pnc", "build", "get", "0" }))
+                                .isEqualTo(1));
+        assertThat(text).contains("JSON command is enabled: false");
     }
 
     @Test
     void testPigTempBuild() throws Exception {
         File pncClasses = new File(App.class.getClassLoader().getResource("").getFile());
         File root = pncClasses.getParentFile().getParentFile().getParentFile();
         File buildConfig = new File(root, FilenameUtils.separatorsToSystem("pig/src/test/resources/empty"));
-
         File configYaml = new File(root, PNC_TEST_CLASSES);
 
-        String text = "";
-        try {
-            text = tapSystemErr(
-                    () -> new App().run(
-                            new String[] {
-                                    "-p",
-                                    configYaml.toString(),
-                                    "-v",
-                                    "pig",
-                                    "configure",
-                                    "--releaseStorageUrl",
-                                    "http://www.example.com",
-                                    "--tempBuild",
-                                    buildConfig.toString() }));
-        } catch (FatalException e) {
-            assertTrue(e.getMessage().contains("Keycloak authentication failed"));
-        }
-        assertFalse(text.contains("Unknown option: '--tempBuild'"));
+        String text = tapSystemErr(
+                () -> new App().run(
+                        new String[] {
+                                "-p",
+                                configYaml.toString(),
+                                "-v",
+                                "pig",
+                                "configure",
+                                "--releaseStorageUrl",
+                                "http://www.example.com",
+                                "--tempBuild",
+                                buildConfig.toString() }));
+
+        assertThat(text).satisfiesAnyOf(
+                t -> assertThat(t).contains("Unknown option: '--tempBuild'"),
+                t -> assertThat(t).contains("Keycloak authentication failed"));
```

Comment:
I think there is something wrong with this test. There is never any exception thrown (`FatalException` or otherwise) even though stderr contains "Keycloak authentication failed" in the exception stack trace in the output (so the `tapSystemErr()` is swallowing it?). I never see the message about unknown option, but if the command line fails to parse, how would it be getting far enough to throw an exception in the first place? What is the intended output of the test?

Reply:
Hmm, yes I see, its both a flaw in the test and the code. Will look into it.

Reply:
Improved in https://github.com/project-ncl/bacon/pull/452